sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -463,12 +463,23 @@ class Analyzer(
         // Find sort attributes that are projected away so we can temporarily add them back in.
         val (newOrdering, missingAttr) = resolveAndFindMissing(ordering, a, groupingRelation)
 
+        val existingNames = aggs.map(_.name)
+        val prefix = {
+          var init = "_aggOrdering"
+          while (existingNames.exists(_.startsWith(init))) {
+            init = "_" + init
+          }
+          init
+        }
+        var i = 0
         // Find aggregate expressions and evaluate them early, since they can't be evaluated in a
         // Sort.
         val (withAggsRemoved, aliasedAggregateList) = newOrdering.map {
-          case aggOrdering if aggOrdering.collect { case a: AggregateExpression => a }.nonEmpty =>
-            val aliased = Alias(aggOrdering.child, "_aggOrdering")()
-            (aggOrdering.copy(child = aliased.toAttribute), Some(aliased))
+          case aggOrdering if aggOrdering.find(_.isInstanceOf[AggregateExpression]).isDefined =>
+            val aliasName = prefix + i
+            i += 1
+            val aliased = Alias(aggOrdering.child, aliasName)()
+            (aggOrdering.copy(child = UnresolvedAttribute(aliasName)), Some(aliased))
```

Comment:
Not a subclass, just another field (like `qualifiers`) that we can use to say that a given attribute produced by an alias can't be resolved by name, but only by explicitly calling `toAttribute` on the alias.